diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt

```diff
@@ -26,44 +28,50 @@ open class RuleConfiguration(protected val config: Map<String, String>)
 object EmptyConfiguration: RuleConfiguration(mapOf())
 
 /**
- * class returns the list of configurations that we have read from a json: rules-config.json
+ * class returns the list of configurations that we have read from a yml: diktat-analysis.yml
  */
 open class RulesConfigReader(override val classLoader: ClassLoader) : JsonResourceConfigReader<List<RulesConfig>>() {
     companion object {
         val log: Logger = LoggerFactory.getLogger(RulesConfigReader::class.java)
     }
 
     override fun parseResource(fileStream: BufferedReader): List<RulesConfig> {
-        val mapper = jacksonObjectMapper()
-        val jsonValue = fileStream.lines().collect(Collectors.joining())
-        return mapper.readValue(jsonValue)
+        val mapper = ObjectMapper(YAMLFactory())
+        mapper.registerModule(KotlinModule())
+        val list: List<RulesConfig> = mapper.readValue(fileStream)
+        fileStream.close()
```

Comment:
You can use kotlin's `use` extension of `Closeable` to close resource and return result,